packages/astro-embed-mastodon/Toot.astro

```diff
@@ -0,0 +1,138 @@
+---
+import PreviewCard, { type MastodonPreviewCard } from './PreviewCard.astro';
+import MediaAttachment, {
+	type MastodonMediaAttachment,
+} from './MediaAttachment.astro';
+import './Toot.css';
+
+export interface Props {
+	id: string;
+}
+const { id } = Astro.props;
+
+/**
+ * Thanks to @astrojs/starlight
+ * https://github.com/withastro/starlight/blob/8a861d16b586b019f61f30d93c61bdcd58e1503f/packages/starlight/utils/i18n.ts#L12
+ *
+ * A list of well-known right-to-left languages used as a fallback when determining the text
+ * direction of a locale is not supported by the `Intl.Locale` API in the current environment.
+ *
+ * @see getLocaleDir()
+ * @see https://en.wikipedia.org/wiki/IETF_language_tag#List_of_common_primary_language_subtags
+ */
+const wellKnownRTL = ['ar', 'fa', 'he', 'prs', 'ps', 'syc', 'ug', 'ur'];
+
+// Regex to extract a toot ID from a Mastodon status URL
+const idRegExp = /(\d+)$/;
+
+// Thanks to @astrojs/starlight
+// https://github.com/withastro/starlight/blob/8a861d16b586b019f61f30d93c61bdcd58e1503f/packages/starlight/utils/i18n.ts#L177-L188
+function getLocaleDir(locale: Intl.Locale): 'ltr' | 'rtl' {
+	if ('textInfo' in locale) {
+		// @ts-expect-error - `textInfo` is not typed but is available in v8 based environments.
+		return locale.textInfo.direction;
+	} else if ('getTextInfo' in locale) {
+		// @ts-expect-error - `getTextInfo` is not typed but is available in some non-v8 based environments.
+		return locale.getTextInfo().direction;
+	}
+	// Firefox does not support `textInfo` or `getTextInfo` yet so we fallback to a well-known list
+	// of right-to-left languages.
+	return wellKnownRTL.includes(locale.language) ? 'rtl' : 'ltr';
+}
+
+function extractID(url: string) {
+	const match = url.match(idRegExp);
+	return match?.[0];
+}
+
+async function fetchToot(id: string) {
+	try {
+		const tootId = extractID(id);
+		if (!tootId) throw new Error('Invalid toot URL');
+		const apiUrl = new URL(`${new URL(id).origin}/api/v1/statuses/${tootId}`);
+		const response = await fetch(apiUrl);
```

Comment:
The other components use a “safe” fetching method from the utils package:

https://github.com/delucis/astro-embed/blob/6fce869729bdc178f8ce71ce6eeddc29f023800f/packages/astro-embed-utils/index.ts#L28-L33

This helps avoid breaking builds and also has a built-in caching layer to speed up repeat fetches.

Reply:
Oh, I totally missed that as I mostly based my work on the Tweet component which does not use it (I guess this would be another PR). Updated the PR to use it. Thanks for pointing it out!